src/i2c.rs

```diff
@@ -451,7 +451,7 @@ macro_rules! i2c {
                 fn clock(clocks: &Clocks) -> Hertz {
                     // NOTE(unsafe) atomic read with no side effects
                     match unsafe { (*RCC::ptr()).cfgr3.read().$i2cXsw().variant() } {
-                        I2C1SW_A::HSI => 8.mhz().into(),
+                        I2C1SW_A::HSI => Hertz(8_000_000),
```

Comment:
Whats the motivation to use Hertz instead of Megahertz here?

Reply:
In this case it is/was the return `type` of that specific function. I didn't want to change it to `MHz` so that I didn't break anything unintentionally.

Reply:
…and using Megahertz and casting it back doesn't really make it more readable.